GAME/BOX.C

```diff
@@ -434,10 +434,36 @@ void CreatureDie(short item_number, int explode)// (F)
 	}
 }
 
-int BadFloor(long x, long y, long z, long box_height, long next_height, int room_number, struct lot_info* LOT)
+int BadFloor(long x, long y, long z, long box_height, long next_height, int room_number, struct lot_info* LOT)// (F)
 {
-	S_Warn("[BadFloor] - Unimplemented!\n");
-	return 0;
+	short room_num = room_number;
```

Comment:
Variable not needed, psx just takes room_number. No additional var.

Reply:
Variable is needed because room_number is an int and GetFloor expects a short*

Reply:
Why not cast? I'd assume that's what they did originally?

floor = GetFloor(x, y, z, (short*) &room_number);

Reply:
That code would break on a big-endian system

It just happens to work because fortunately PSX and PC use little endian

Reply:
Approved.